File: modelo/Envio.php
<?php

require_once 'EnvioDAO.php'; // Certifique-se de que o caminho está correto

class Envio {

    private $clienteid;
    private $cameraid;
    private $acesso;
    private $data_entrega;
    private $data_envio;
    private $data_instalacao;
    private $nota_fiscal;
    private $sequencia;
    private $numero_pedido;
    private $status;
    private $envioDAO;

    // Construtor Default
    public function __construct($clienteid = 0, $cameraid = 0, $acesso = "", $data_entrega = "", $data_envio = "", $data_instalacao = "", $nota_fiscal = 0, $sequencia = 0, $numero_pedido = 0, $status = "") {
        $this->clienteid = $clienteid;
        $this->cameraid = $cameraid;
        $this->acesso = $acesso;
        $this->data_entrega = $data_entrega;
        $this->data_envio = $data_envio;
        $this->data_instalacao = $data_instalacao;
        $this->nota_fiscal = $nota_fiscal;
        $this->sequencia = $sequencia;
        $this->numero_pedido = $numero_pedido;
        $this->status = $status;
        $this->envioDAO = new EnvioDAO();
    }

    // Gets
    public function getEnvioid($clienteid, $cameraid) {
        return $this->envioDAO->getEnvioidDAO($clienteid, $cameraid);
    }

    public function getClienteid() {
        return $this->envioDAO->getClienteidDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getClienteID($ID) {
        return $this->envioDAO->getClienteidDAO($ID);
    }

    public function getCameraid() {
        return $this->envioDAO->getCameraidDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getCameraID($ID) {
        return $this->envioDAO->getCameraidDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getAcesso() {
        return $this->envioDAO->getAcessoDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getAcessoID($ID) {
        return $this->envioDAO->getAcessoDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getData_Entrega() {
        return $this->envioDAO->getData_EntregaDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getData_EntregaID($ID) {
        return $this->envioDAO->getData_EntregaDAO($ID);
    }

    public function getData_Envio() {
        return $this->envioDAO->getData_EnvioDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getData_EnvioID($ID) {
        return $this->envioDAO->getData_EnvioDAO($ID);
    }

    public function getData_Instalacao() {
        return $this->envioDAO->getData_InstalacaoDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getData_InstalacaoID($ID) {
        return $this->envioDAO->getData_InstalacaoDAO($ID);
    }

    public function getNotaFiscal() {
        return $this->envioDAO->getNotaFiscalDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getNotaFiscalID($ID) {
        return $this->envioDAO->getNotaFiscalDAO($ID);
    }

    public function getSequencia() {
        return $this->envioDAO->getSequenciaDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getSequenciaID($ID) {
        return $this->envioDAO->getSequenciaDAO($ID);
    }

    public function getNumero_Pedido() {
        return $this->envioDAO->getNumero_PedidoDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getNumero_PedidoID($ID) {
        return $this->envioDAO->getNumero_PedidoDAO($ID);
    }

    public function getStatus() {
        return $this->envioDAO->getStatusDAO($this->getEnvioid($this->clienteid, $this->cameraid));
    }

    public function getStatusID($ID) {
        return $this->envioDAO->getStatusDAO($ID);
    }

    // Get lista de envios
    public function getEnvios() {
        return $this->envioDAO->getEnviosDAO();
    }

    // Sets
    public function setClienteid($clienteid) {
        $this->envioDAO->setClienteidDAO($this->getEnvioid($this->clienteid, $this->cameraid), $clienteid);
        $this->clienteid = $clienteid;
    }

    public function setCameraid($cameraid) {
        $this->envioDAO->setCameraidDAO($this->getEnvioid($this->clienteid, $this->cameraid), $cameraid);
        $this->cameraid = $cameraid;
    }

    public function setAcesso($acesso) {
        $this->envioDAO->setAcessoDAO($this->getEnvioid($this->clienteid, $this->cameraid), $acesso);
        $this->acesso = $acesso;
    }

    public function setAcessoID($ID, $acesso) {
        $this->envioDAO->setAcessoDAO($ID, $acesso);
    }

    public function setData_Entrega($data_entrega) {
        $this->envioDAO->setData_EntregaDAO($this->getEnvioid($this->clienteid, $this->cameraid), $data_entrega);
        $this->data_entrega = $data_entrega;
    }

    public function setData_EntregaID($ID, $data_entrega) {
        $this->envioDAO->setData_EntregaDAO($ID, $data_entrega);
    }

    public function setData_Envio($data_envio) {
        $this->envioDAO->setData_EnvioDAO($this->getEnvioid($this->clienteid, $this->cameraid), $data_envio);
        $this->data_envio = $data_envio;
    }

    public function setData_EnvioID($ID, $data_envio) {
        $this->envioDAO->setData_EnvioDAO($ID, $data_envio);
    }

    public function setData_Instalacao($data_instalacao) {
        $this->envioDAO->setData_InstalacaoDAO($this->getEnvioid($this->clienteid, $this->cameraid), $data_instalacao);
        $this->data_instalacao = $data_instalacao;
    }

    public function setData_InstalacaoID($ID, $data_instalacao) {
        $this->envioDAO->setData_InstalacaoDAO($ID, $data_instalacao);
    }

    public function setNota_Fiscal($nota_fiscal) {
        $this->envioDAO->setNota_FiscalDAO($this->getEnvioid($this->clienteid, $this->cameraid), $nota_fiscal);
        $this->nota_fiscal = $nota_fiscal;
    }

    public function setSequencia($sequencia) {
        $this->envioDAO->setSequenciaDAO($this->getEnvioid($this->clienteid, $this->cameraid), $sequencia);
        $this->sequencia = $sequencia;
    }

    public function setNumero_Pedido($numero_pedido) {
        $this->envioDAO->setNumero_PedidoDAO($this->getEnvioid($this->clienteid, $this->cameraid), $numero_pedido);
        $this->numero_pedido = $numero_pedido;
    }

    public function setStatus($ID, $status) {
        $this->envioDAO->setStatusDAO($ID, $status);
        $this->status = $status;
    }

    // Adiciona Envio ao banco de dados
    public function addEnvio($clienteid, $cameraid, $nota_fiscal, $sequencia, $numero_pedido, $status) {
        $this->envioDAO->addEnvioDAO($clienteid, $cameraid, $nota_fiscal, $sequencia, $numero_pedido, $status);
    }

    // Remove Envio do banco de dados
    public function removeEnvio($envioid) {
        $this->envioDAO->removeEnvioDAO($envioid);
    }
}

?>

File: Model/envio.py
from Controller.envioDAO import EnvioDAO

class Envio:

    def __init__(self, clienteid=0, cameraid=0, acesso="", nota_fiscal=0, sequencia=0, numero_pedido=0, status=""):
        self.clienteid = clienteid
        self.cameraid = cameraid
        self.acesso = acesso
        self.nota_fiscal = nota_fiscal
        self.sequencia = sequencia
        self.numero_pedido = numero_pedido
        self.status = status
        self.envio_dao = EnvioDAO()

    ''' GETS '''
    def getClienteid(self, envioid):
        return self.envio_dao.getClienteidDAO(envioid)
    
    def getCameraid(self, envioid):
        return self.envio_dao.getCameraidDAO(envioid)
    
    def getAcesso(self, envioid):
        return self.envio_dao.getAcessoDAO(envioid)
    
    def getNotaFiscal(self, envioid):
        return self.envio_dao.getNotaFiscalDAO(envioid)
    
    def getSequencia(self, envioid):
        return self.envio_dao.getSequenciaDAO(envioid)
    
    def getNumeroPedido(self, envioid):
        return self.envio_dao.getNumeroPedidoDAO(envioid)
    
    def getStatus(self, envioid):
        return self.envio_dao.getStatusDAO(envioid)

    ''' SETS '''
    def setClienteid(self, novo_clienteid, envioid):
        self.clienteid = novo_clienteid
        self.envio_dao.setClienteidDAO(novo_clienteid, envioid)
    
    def setCameraid(self, nova_cameraid, envioid):
        self.cameraid = nova_cameraid
        self.envio_dao.setCameraidDAO(nova_cameraid, envioid)
    
    def setAcesso(self,  novo_acesso, envioid):
        self.acesso =  novo_acesso
        self.envio_dao.setAcessoDAO(novo_acesso, envioid)
    
    def setNotaFiscal(self, nova_notaFiscal, envioid):
        self.nota_fiscal = nova_notaFiscal
        self.envio_dao.setNotaFiscalDAO(nova_notaFiscal, envioid)
    
    def setSequencia(self, nova_sequencia, envioid):
        self.sequencia = nova_sequencia
        self.envio_dao.setSequenciaDAO(nova_sequencia, envioid)
    
    def setNumeroPedido(self, novo_numeropedido, envioid):
        self.numero_pedido = novo_numeropedido
        self.envio_dao.setNumeroPedidoDAO(novo_numeropedido, envioid)
    
    def setStatus(self, novo_status, envioid):
        self.status = novo_status
        self.envio_dao.setStatusDAO(novo_status, envioid)

    # Adiciona Envio
    def add_envio(self, clienteid, cameraid, nota_fiscal, sequencia, numero_pedido):
        return self.envio_dao.addEnvioDAO(clienteid, cameraid, nota_fiscal, sequencia, numero_pedido)

    def visualizar_dados(self):
        return self.envio_dao.visualizarDadosDAO()

    # Edita Envio 
    def edit_envio(self, envioid, clienteid, cameraid, acesso, nota_fiscal, sequencia, numero_pedido, status):
        self.setClienteid(clienteid, envioid)
        self.setCameraid(cameraid, envioid)
        self.setAcesso(acesso, envioid)
        self.setNotaFiscal(nota_fiscal, envioid)
        self.setSequencia(sequencia, envioid)
        self.setNumeroPedido(numero_pedido, envioid)
        self.setStatus(status, envioid)

    # Remove Envio
    def delete_envio(self, envio_id):
        self.envio_dao.setStatusDAO("Finalizado", envio_id)